ibis/backends/datafusion/__init__.py

```diff
@@ -183,7 +181,46 @@ def execute(
                 # dummy datafusion table
                 compiled = self.compile(expr, params, **kwargs)
                 frame = self._context.empty_table().select(compiled)
-            table = _to_pyarrow_table(frame)
+            return frame
+        else:
+            raise com.IbisError(
+                f"Cannot execute expression of type: {type(expr)}"
+            )
+
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+        **kwargs: Any,
+    ) -> Iterable[pa.RecordBatch]:
+        frame = self._get_frame(expr, params, limit, **kwargs)
+        yield from frame.collect()
+
+    def to_pyarrow(
```

Comment:
This shouldn't be necessary by having a default implementation for `to_pyarrow()`